### ISSUE TITLE
Example of too demanding 1-letter `enum_variant_names` warning

### ISSUE THREAD

**Issue by NikVolf** [Manishearth/rust-clippy#665]
```
src/evm/tests.rs:28:1: 30:2 warning: All variants have the same prefix: `C`, #[warn(enum_variant_names)] on by default
src/evm/tests.rs:28 enum FakeCallType {
src/evm/tests.rs:29     CALL, CREATE
src/evm/tests.rs:30 }
```

**Comment by Manishearth** [comment]
cc @oli-obk 

looks like a straightforward fix.

**Comment by mcarton** [comment]
Does this lint because it’s all uppercase or does the lint ignores case?
```rust
enum FakeCallType {
    Create, Cry, Cryogenize
}
```
Looks like there is no tests for that lint.

**Comment by Manishearth** [comment]
The lint looks for CamelCase prefixes (i.e. if all variants are CamelFoo for different Foos). The way it's currently written, the C in CALL is a valid Camel :smile:

**Comment by mcarton** [comment]
I should have asked ”Does this lint because it’s all 1-character :camel:?” :smile:.

**Comment by oli-obk** [comment]
xD whoops, yea should be easy to fix

**Comment by oli-obk** [comment]
What about things like `enum Something { CCall, CCreate, CCryogenize }` ?

**Comment by mcarton** [comment]
That one should probably warn.
